processor/src/memory/mod.rs

```diff
@@ -221,15 +221,23 @@ impl Memory {
 
     /// Returns all the addresses and values stored in memory.
     pub fn get_all_values(&self) -> Vec<(u64, Word)> {
-        self.get_values(RangeInclusive::new(0, u64::MAX))
+        self.get_values(RangeInclusive::new(0, u64::MAX), None)
     }
 
-    /// Returns values within a range of addresses.
-    pub fn get_values(&self, range: RangeInclusive<u64>) -> Vec<(u64, Word)> {
+    /// Returns values within a range of addresses, or optionally all values
+    /// at a partciular step. If no step is specified, then values at the last
+    /// step is returned.
+    pub fn get_values(&self, range: RangeInclusive<u64>, step: Option<usize>) -> Vec<(u64, Word)> {
```

Comment:
I'd prefer keeping this method as it was and just add a new method `get_values_at()`. Than `get_values()` could internally call `get_values_at()` and pass the current clock cycle.